### ISSUE TITLE
FileNotFoundError for QuantumESPRESSO with hdf5

### ISSUE THREAD

**Issue by JulVandenBroeck** [band-unfolding/banduppy#32]
Hello!

I'd really like to use your package on our HPC infrastructure. However, QuantumESPRESSO is compiled using the hdf5 flag, which leads to the following error when loading the band structure data:

```
Traceback (most recent call last):
  File "/user/gent/437/vsc43727/quantum_espresso/.venv/lib/python3.9/site-packages/irrep/kpoint.py", line 923, in __init_espresso
    fWFC=FF("{}.save/{}.dat".format(prefix,wfcname.lower()),"r")
  File "/user/gent/437/vsc43727/quantum_espresso/.venv/lib/python3.9/site-packages/scipy/io/_fortran.py", line 123, in __init__
    self._fp = open(filename, '%sb' % mode)
FileNotFoundError: [Errno 2] No such file or directory: 'si.save/wfc1.dat'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/user/gent/437/vsc43727/quantum_espresso/si/bands/step4/unfold.py", line 37, in <module>
    bands = banduppy.BandStructure(code="espresso", prefix="si", EF="6.6416")
  File "/user/gent/437/vsc43727/quantum_espresso/.venv/lib/python3.9/site-packages/irrep/bandstructure.py", line 152, in __init__
    self.__init_espresso(
  File "/user/gent/437/vsc43727/quantum_espresso/.venv/lib/python3.9/site-packages/irrep/bandstructure.py", line 924, in __init_espresso
    kp = Kpoint(
  File "/user/gent/437/vsc43727/quantum_espresso/.venv/lib/python3.9/site-packages/irrep/kpoint.py", line 215, in __init__
    self.WF, self.ig = self.__init_espresso(
  File "/user/gent/437/vsc43727/quantum_espresso/.venv/lib/python3.9/site-packages/irrep/kpoint.py", line 925, in __init_espresso
    fWFC=FF("{}.save/{}.dat".format(prefix,wfcname.upper()),"r")
  File "/user/gent/437/vsc43727/quantum_espresso/.venv/lib/python3.9/site-packages/scipy/io/_fortran.py", line 123, in __init__
    self._fp = open(filename, '%sb' % mode)
FileNotFoundError: [Errno 2] No such file or directory: 'si.save/WFC1.dat'
```

It seems that the package expects ".dat" files, but this version of QE generates "wfc*.hdf5" files. Would be really nice to see a fix for this!

Thanks in advance!

Below is my python file:
<details>
<summary>Python script</summary>

```python
import os
current_dir = os.path.dirname(__file__)

import numpy as np
import banduppy


# Conventional cell
super_cell_size = np.diag([1,1,1])

# Path in Brillouin zone: L-G-X-U, K-G
k_path = [[1/2,1/2,1/2], [0,0,0], [1,0,0], [1,1/4,1/4], None, [3/4,3/4,0], [0,0,0]]

# Point labels
k_point_labels = "LGXUKG"

# Number of k-points per path segments
num_points_per_path_seg = (20, 30, 10, 30)

# Weight of the k-points
k_weights = 1

# Save the SC k-points to a file
save_to_file = True
save_to_dir = os.path.join(current_dir, 'outdir')
file_format = 'qe'


# Initiate the unfolding method
band_unfold = banduppy.Unfolding(supercell=super_cell_size, print_log='high')

# Read the band structure
cwd = os.getcwd()
out_dir = os.environ.get("ESPRESSO_TMPDIR", ".")
os.chdir(os.path.join(current_dir, out_dir))
bands = banduppy.BandStructure(code="espresso", prefix="si", EF="6.6416")
os.chdir(cwd)
```
</details>

**Comment by stepan-tsirkin** [comment]
Hi @JulVandenBroeck ,

`banduppy.BandStructure` is essentially a shortcut to `irrep.bandstructure.BandStructure` and hdf5 support has been added to irrep recently: 

https://github.com/irreducible-representations/irrep/blob/e47d559fe481d67d437f0c8d4c0b47ac207117b3/irrep/readfiles.py#L704-L709

So update irrep to latest version [1.10.0](https://pypi.org/project/irrep/1.10.0/#history) and it should work. If not - tell me.

**Comment by JulVandenBroeck** [comment]
Thanks for the quick response!

It seems that banduppy prevents the use of irrep version above 1.8.3:
```
Traceback (most recent call last):
  File "/user/gent/437/vsc43727/quantum_espresso/si/bands/step4/unfold.py", line 5, in <module>
    import banduppy
  File "/user/gent/437/vsc43727/quantum_espresso/.venv/lib/python3.9/site-packages/banduppy/__init__.py", line 6, in <module>
    raise ImportError("Critical bugs found in irrep-1.6.1 and >1.8.3 versions. We request BandUPpy users to use irrep-1.8.3 until the next release of the irrep package.")
ImportError: Critical bugs found in irrep-1.6.1 and >1.8.3 versions. We request BandUPpy users to use irrep-1.8.3 until the next release of the irrep package.
```

**Comment by JulVandenBroeck** [comment]
If i comment out the exception in the __init__.py file, I get an error in irrep itself:

```
Traceback (most recent call last):
  File "/user/gent/437/vsc43727/quantum_espresso/si/bands/step4/unfold.py", line 5, in <module>
    import banduppy
  File "/user/gent/437/vsc43727/quantum_espresso/.venv/lib/python3.9/site-packages/banduppy/__init__.py", line 8, in <module>
    from  irrep.bandstructure import BandStructure
  File "/user/gent/437/vsc43727/quantum_espresso/.venv/lib/python3.9/site-packages/irrep/bandstructure.py", line 27, in <module>
    from .spacegroup import SpaceGroup
  File "/user/gent/437/vsc43727/quantum_espresso/.venv/lib/python3.9/site-packages/irrep/spacegroup.py", line 25, in <module>
    from irreptables import IrrepTable
ModuleNotFoundError: No module named 'irreptables'
```

I guess I have to move the issue to irrep...

**Comment by JulVandenBroeck** [comment]
It seems that my packages were broken. After removing irrep and irreptables, then installing irreptables first and irrep second, everything seems to work fine!

Thanks for the help :)

**Comment by bmondal94** [comment]
Hi @JulVandenBroeck ,

We have just released BandupPy latest version (v-0.3.4). 

The new version includes: 
1. irrep latest version support
2. Major upgrade of effective mass calculation function
3. Added quantum espresso tutorial

Feel free to upgrade Banduppy to the latest version at your convenience.
